### DISCUSSION TITLE
How to enable JS to use TWE like carousel

### DISCUSSION THREAD

**Discussion by webmasterpf** [mdbootstrap/TW-Elements#1679] [Support from community]
Hello,
I'm Unable to initiate variables for JS TWE like carousel or button.
I use webpack to compile TW css and assets. I'm new with this technology so I miss something perhaps.

**To Reproduce**
Create a JS file with this code in it. This file is loaded via the librarie theme file from drupal.
```
// Initialization for ES Users
import {
    Carousel,
    Ripple,
    initTE,
  } from "tw-elements";
/* Caroussel Tailwind*//* Effets sur boutons*/
initTE({ Carousel, Ripple });
```

**Expected behavior**
Loading JS for TWE into a drupal theme.

**Actual behavior**
Uncaught:  import statement outside a module (at tailwind-custom.js?v=9.5.9:2:1)     

Thanks

**Comment by juujisai** [comment]
Hi, im not sure how you are importing the js file into drupal theme, but you can check if you are importing the file as a `type="module"`. For example:
`<script type="module" src="./my.js"></script>`

**Comment by webmasterpf** [comment]
Hi
Usually , to import into drupal theme, I do this in the .librarie as described here: https://www.drupal.org/docs/drupal-apis/javascript-api/add-javascript-to-your-theme-or-module

Actually this result to:
`<script src="/themes/custom/starter-d9-2022/js/tailwind-custom.js?v=9.5.9"></script>`
So I need to find the way to specify  `type="module"`

**Comment by juujisai** [comment]
Maybe something like this could work? 
```
custom-js:
  js:
    js/tw-elements.js: { type: module }
```

**Comment by webmasterpf** [comment]
I tried it few days ago , and I face off an error: `Exception: Invalid JS asset type. in Drupal\Core\Asset\JsCollectionRenderer->render() (line 105 of core/lib/Drupal/Core/Asset/JsCollectionRenderer.php).` https://drupal.slack.com/archives/C1BB308HH/p1687177185737069?thread_ts=1687171176.996129&cid=C1BB308HH  :/
